lib/moonshot/controller_config.rb

```diff
@@ -8,23 +8,28 @@ class ControllerConfig
     attr_accessor :additional_tag
     attr_accessor :app_name
     attr_accessor :artifact_repository
+    attr_accessor :answer_file
     attr_accessor :build_mechanism
     attr_accessor :deployment_mechanism
     attr_accessor :environment_name
+    attr_accessor :interactive
     attr_accessor :interactive_logger
     attr_accessor :parent_stacks
     attr_accessor :plugins
     attr_accessor :show_all_stack_events
-    attr_accessor :parameter_strategy
     attr_accessor :ssh_config
     attr_accessor :ssh_command
     attr_accessor :ssh_auto_scaling_group_name
     attr_accessor :ssh_instance
     attr_accessor :project_root
+    attr_accessor :parameters
+    attr_accessor :parameter_overrides
 
     def initialize
+      @interactive           = true
       @interactive_logger    = InteractiveLogger.new
-      @parameter_strategy    = Moonshot::ParameterStrategy::DefaultStrategy.new
+      @parameters            = {}
+      @parameter_overrides   = {}
```

Comment:
Addressed in latest patch by using a real factory method and sane state in the class.